### DISCUSSION TITLE
Маршруты и POI на карте

### DISCUSSION THREAD

**Discussion by FelixPeroff** [0niel/university-app#220] [Идеи]
_Одна из не менее важных функций приложения - это помощь в построении маршрутов от одной аудитории до другой, так как первокурсникам, особенно сейчас, крайне тяжело ориентироваться в кампусе ввиду отсутствия нормального Факела_

Предлагаю сделать приоритетной задачу добавления маршрутов на карту внутри приложения, чтобы можно было из приложения получить информацию о том, **где находится** та или иная аудитория **по поиску**, а также узнать **как** до неё **дойти**.

<img width="648" alt="image" src="https://user-images.githubusercontent.com/47608978/131262021-25c5b379-38ad-4f70-97f8-cf7617444060.png">

В приложении на экране "Карта" появится кнопка "маршруты", по нажатию на которую откроется экран построения маршрута. 
По умолчанию отправной точкой всех маршрутов должна быть аудитория последней прошедшей пары (той, которая идёт в данный момент, или той, которая только закончилась, если перемена). В качестве конечной точки можно указать аудиторию (А-108, Г-304, …), а также POI (места интереса, например "столовая", "буфет", "туалет", "медпункт". Если таких мест несколько, выбирается ближайшее). Полагаю, что потребуется создать базу данных для занесения в неё POI, готов заняться этим. 

Замечание: есть несколько контекстных POI, например "туалет" или "учебный отдел", поскольку такие места могут отличаться для каждого студента. При выборе в качестве точки назначения "туалета", будет учитываться пол студента на основании профиля (который на момент написания ещё делается, но в профиле нужно будет указать пол, и тогда приложение будет подсказывать соответственно ближайший женский или мужской туалет). При выборе "учебного отдела" в качестве точки назначения, будет учитываться институт студента (его можно узнать из первой буквы инициалов его группы (**И**кбо, **К**рбо, **И**вбо)

**Comment by FelixPeroff** [comment]
Готовая реализация навигации внутри кампуса 78
https://ischemes.ru/group/rtu-mirea/vern78

**Comment by Dragonprod** [comment]
Реализовано на https://map.mirea.ru/
